src/validation.cpp

```diff
@@ -848,8 +848,11 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         return false; // state filled in by CheckTxInputs
     }
 
-    if (m_pool.m_opts.require_standard && !AreInputsStandard(tx, m_view)) {
-        return state.Invalid(TxValidationResult::TX_INPUTS_NOT_STANDARD, "bad-txns-nonstandard-inputs");
+    if (m_pool.m_opts.require_standard) {
+        state = HasNonStandardInput(tx, m_view);
```

Comment:
Seems like it'd be nicer to just pass `state` into the function rather than creating a new one and copying it.

Reply:
> Seems like it'd be nicer to just pass state into the function 

I am taking this review comment as a non-blocking (nit) suggestion based on the tone.

I will update this when there are blocking comments that indicate a need to retouch the diffs.

Also, I would like to know whether this PR is conceptually acknowledged, @luke-jr.
Thanks for your review.

Reply:
In this case, a new state is indeed created in `HasNonStandardInput`, but either RVO or an implicit move is performed not a copy?

I think, in general, returning a value is preferred over using out parameters?
same was also recommended here https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c-functions-and-methods